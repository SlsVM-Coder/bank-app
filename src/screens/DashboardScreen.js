import React from 'react';
import { useSelector } from 'react-redux';
import { BalanceCard } from './../components/BalanceCard';

export const DashboardScreen = () => {
  const { name } = useSelector((state) => state.auth);

  return (
    <div>
      <h1>Welcome back, {name}</h1>
      <p>You can:</p>
      <p>Check your money, transfer money or view your last movements. </p>
      <BalanceCard />
    </div>
  );
};
