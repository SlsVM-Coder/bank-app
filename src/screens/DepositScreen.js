import React from 'react';
import { useForm } from './../hooks/useForm';
import { depositMoney } from './../actions/money';
import { useDispatch } from 'react-redux';

export const DepositScreen = () => {
  const dispatch = useDispatch();
  const [formValues, handleInputChange] = useForm({
    quantity: '',
  });
  const { quantity } = formValues;

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(depositMoney(parseFloat(quantity)));
  };

  return (
    <div>
      <h1>deposit</h1>
      <form onSubmit={handleSubmit}>
        <label htmlFor="quantity">Quantity:</label>
        <input
          type="number"
          placeholder="0"
          name="quantity"
          id="quantity"
          className="input-text"
          value={quantity}
          onChange={handleInputChange}
        />
      </form>
    </div>
  );
};
