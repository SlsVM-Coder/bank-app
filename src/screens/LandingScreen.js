import React from 'react';
import { Link } from 'react-router-dom';
import { Layout } from '../components/Layout';

export const LandingScreen = () => {
  return (
    <Layout
      component="Landing"
      title="Your favourite online wallet"
      img="landingImg"
    >
      <Link to="/auth">
        <button renderas="button" className="btn-primary">
          <span>Get started</span>
        </button>
      </Link>
    </Layout>
  );
};
