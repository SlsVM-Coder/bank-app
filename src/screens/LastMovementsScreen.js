import React from 'react';
import { useSelector } from 'react-redux';
import { Movement } from './../components/Movement';
import { Layout } from './../components/Layout';

export const LastMovementsScreen = () => {
  const { movements } = useSelector((state) => state.money);

  return (
    <Layout
      component="LastMovements"
      img="actionsImg"
      title="Last movements"
    >
      <div className="MovementsContainer">
        {movements.length > 0 ? (
          movements.map((movement, index) => (
            <Movement
              key={`${index}-${movement.date}`}
              {...movement}
            />
          ))
        ) : (
          <p>You don't have any movements yet.</p>
        )}
      </div>
    </Layout>
  );
};
