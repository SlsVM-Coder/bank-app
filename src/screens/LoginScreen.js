import React from 'react';
import { Link } from 'react-router-dom';
import { LoginForm } from './../components/LoginForm';
import { Layout } from './../components/Layout';

export const LoginScreen = () => {
  return (
    <Layout component="Login" img="authImg" title="Sign in">
      <LoginForm />
      <div className="otherAuth">
        <p>Don't have an account?</p>
        <Link to="/auth/register" className="link">
          Sign up
        </Link>
      </div>
    </Layout>
  );
};
