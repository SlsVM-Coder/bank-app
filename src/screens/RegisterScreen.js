import React from 'react';
import { Link } from 'react-router-dom';
import { Layout } from '../components/Layout';
import { RegisterForm } from './../components/RegisterForm';

export const RegisterScreen = () => {
  return (
    <Layout
      component="Register"
      img="authImg"
      title="Create an account"
    >
      <RegisterForm />
      <div className="otherAuth">
        <p>Already registered?</p>
        <Link to="/auth/login" className="link">
          Sign in
        </Link>
      </div>
    </Layout>
  );
};
