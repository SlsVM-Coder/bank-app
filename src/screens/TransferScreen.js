import React from 'react';
import { useForm } from './../hooks/useForm';
import { transferMoney } from './../actions/money';
import { useDispatch, useSelector } from 'react-redux';
import { Layout } from './../components/Layout';
import Swal from 'sweetalert2';

export const TransferScreen = () => {
  const dispatch = useDispatch();
  const { email: userEmail } = useSelector((state) => state.auth);
  const { balance } = useSelector((state) => state.money);

  const [formValues, handleInputChange, reset] = useForm({
    quantity: '',
    receiverEmail: '',
  });
  const { quantity, receiverEmail } = formValues;

  const handleSubmit = (e) => {
    e.preventDefault();
    reset();
    if (receiverEmail === userEmail) {
      Swal.fire('Error', "You can't transfer yourself money.", 'error');
      return;
    }
    if (quantity > balance) {
      Swal.fire(
        'Error',
        `You don't have that amount of money. You currently have ${balance}€ in your account.`,
        'error',
      );
      return;
    }
    if (quantity < 0) {
      Swal.fire('Error', 'Quantity must be greater than 0€.', 'error');
      return;
    }

    dispatch(transferMoney(parseFloat(quantity), receiverEmail));
  };

  return (
    <Layout component="Transfer" img="actionsImg" title="Transfer money">
      {typeof balance === 'number' && (
        <p>
          You currently have <b>{balance}€</b> in your account.
        </p>
      )}
      <form onSubmit={handleSubmit}>
        <label htmlFor="quantity">Quantity:</label>
        <input
          type="number"
          placeholder="0€"
          name="quantity"
          id="quantity"
          min="0"
          className="input-text"
          value={quantity}
          onChange={handleInputChange}
        />
        <label htmlFor="receiverEmail">Email of the receiver:</label>
        <input
          type="email"
          placeholder="Email"
          name="receiverEmail"
          id="receiverEmail"
          className="input-text"
          value={receiverEmail}
          onChange={handleInputChange}
        />
        <button className="btn-primary">Transfer</button>
      </form>
    </Layout>
  );
};
