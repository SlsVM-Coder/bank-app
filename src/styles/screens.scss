.Dashboard {
  .ButtonsContainer {
    display: flex;
    flex-direction: column;
    margin-top: 20px;

    button {
      width: 200px;
      margin-bottom: 5px;
    }
  }
}

.LastMovements {
  .MovementsContainer {
    max-height: 800px;
    overflow: auto;
  }
}

.otherAuth {
  margin-top: 10px;
  display: flex;
  align-items: center;

  .link {
    margin-left: 10px;
  }
}

.msgError {
  width: 250px;
  background-color: #ab0d0d;
  color: white;
  padding: 10px 0;
  border-radius: $border-radius;
  padding-left: 10px;
  margin: 10px 0 20px 0;
}
