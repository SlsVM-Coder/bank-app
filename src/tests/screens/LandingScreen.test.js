import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { LandingScreen } from './../../screens/LandingScreen';
import { mount } from 'enzyme';

describe('Pruebas en <LandingScreen />', () => {
  const wrapper = mount(
    <MemoryRouter>
      <LandingScreen />
    </MemoryRouter>,
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
