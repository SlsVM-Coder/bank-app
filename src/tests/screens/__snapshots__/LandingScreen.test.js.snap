// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LandingScreen /> debe de mostrarse correctamente 1`] = `
<div
  className="Container Landing"
>
  <div
    className="LeftContent"
  >
    <div
      className="Content"
    >
      <h1>
         
        Your favourite online wallet
         
      </h1>
      <a
        href="/auth"
        onClick={[Function]}
      >
        <button
          className="btn-primary"
          renderas="button"
        >
          <span>
            Get started
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="RightContent"
  >
    <img
      alt="landingImg"
      src="landingImg.png"
    />
  </div>
</div>
`;
